test/hw3/test_task3.py

```diff
@@ -0,0 +1,36 @@
+from All_home_works.hw3.task3 import Filter, make_filter, sample_data
+
+
+def test_filter():
+    expected = list(range(2, 100, 2))
+    positive_even = Filter(
+        lambda a: a % 2 == 0, lambda a: a > 0, lambda a: isinstance(a, int)
+    )
+    result = positive_even.apply(range(99))
+    assert expected == result
+
+
+def test_filter_case__data_with_negative():
```

Comment:
What is negative?  This name does not describe the purpose of this test well.

Reply:
Rename: test_filter__positive_even__data_with_negative_numbers_in_data